test/spec/automatic/tagspec.js

```diff
@@ -77,10 +77,232 @@ describe("Cloudinary.ImageTag", function() {
+        Object.assign({}, commonImageOptions, {srcset: srcsetBreakpoint})
+    ).toHtml();
+
+    expect(expectedTag).toBe(tagByParams);
+    return expect(expectedTag).toBe(tagByBreakpoint);
+  });
+
+  it("Should support custom transformation for srcset items", function () {
+    let srcsetParams = Object.assign({}, commonSrcset, {
+      transformation: {'crop': 'crop', 'width': 10, 'height': 20},
+    });
+    let customTransformationStr = 'c_crop,h_20,w_10';
+
+    let expectedTag = expectedImageTagFromParams(
+        fullPublicId,
+        commonTransformationStr,
+        customTransformationStr,
+        breakpointList
+    );
+    let tag = new cloudinary.ImageTag(
+        fullPublicId,
+        Object.assign({}, commonImageOptions, {srcset: srcsetParams})
+    ).toHtml();
+
+    return expect(expectedTag).toBe(tag);
+  });
+
+  it("Should populate sizes attribute", function () {
+    let srcsetParams = Object.assign({}, commonSrcset, {
+      sizes: true,
+    });
+    let expectedSizes = [];
+    for(let key in breakpointList){
+      if (breakpointList.hasOwnProperty(key)) {
+        let w = breakpointList[key];
+        expectedSizes.push(`(max-width: ${w}px) ${w}px`);
+      }
+    }
+    let attributes = {"sizes": expectedSizes.join(', ')};
+
+    let expectedTag = expectedImageTagFromParams(
+        fullPublicId,
+        commonTransformationStr,
+        '',
+        breakpointList,
+        attributes
+    );
+    let tag = new cloudinary.ImageTag(
+        fullPublicId,
+        Object.assign({}, commonImageOptions, {srcset: srcsetParams})
+    ).toHtml();
+
+    return expect(expectedTag).toBe(tag);
+  });
+
+  it("Should support srcset string value", function () {
+    let raw_srcset_value = 'some srcset data as is';
+    let attributes = {'srcset': raw_srcset_value};
+
+    let expectedTag = expectedImageTagFromParams(
+        fullPublicId,
+        commonTransformationStr,
+        '',
+        '',
+        attributes
+    );
+    let legacyTag = new cloudinary.ImageTag(
+        fullPublicId,
+        Object.assign({}, commonImageOptions, {srcset: raw_srcset_value})
+    ).toHtml();
+    let tag = new cloudinary.ImageTag(
+        fullPublicId,
+        Object.assign({}, commonImageOptions, {attributes: attributes})
+    ).toHtml();
+
+    expect(expectedTag).toBe(tag);
+    return expect(expectedTag).toBe(legacyTag);
+  });
+
+  it("Should remove width and height attributes in case srcset is specified", function () {
+    let tag = new cloudinary.ImageTag(
+        fullPublicId,
+        Object.assign({}, commonImageOptions, {
+          width: 500,
+          height: 500,
+          srcset: commonSrcset
+        })
+    ).toHtml();
+    let expectedTag = expectedImageTagFromParams(
+        fullPublicId,
+        commonTransformationStr,
+        '',
+        breakpointList
+    );
+
+    return expect(expectedTag).toBe(tag);
+  });
+
+  it("Should raise ValueError on invalid values", function () {
```

Comment:
In the new commit we use dynamic tests. It allows a single test to fail without duplicating boilerplate code. To our mind  it's easier to understand the code once for a list of various input params rather than reading each test and visually diffing it with a previous one to check whether they only have different params or different checks as well.